cromper/Dockerfile

```diff
@@ -0,0 +1,191 @@
+    binutils-aarch64-linux-gnu \
+    binutils-arm-none-eabi \
+    binutils-djgpp \
+    binutils-mingw-w64-i686 \
+    binutils-mips-linux-gnu \
+    binutils-powerpc-linux-gnu \
+    binutils-sh-elf \
+    cpp \
+    dj64 \
+    dos2unix \
+    dosemu2 \
+    gcc-mips-linux-gnu \
+    iptables \
+    libarchive-tools \
+    libc6-dev-i386 \
+    libdevmapper1.02.1 \
+    libgpgme11 \
+    libnl-route-3-200 \
+    libprotobuf-dev \
+    libtinfo6 \
+    netcat-traditional \
+    unzip \
+    wget \
+    wine \
+    wine32:i386 \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN wget http://security.ubuntu.com/ubuntu/pool/universe/n/ncurses/libtinfo5_6.3-2ubuntu0.1_amd64.deb \
+    && apt install ./libtinfo5_6.3-2ubuntu0.1_amd64.deb \
+    && rm libtinfo5_6.3-2ubuntu0.1_amd64.deb
+
+COPY --from=nsjail /nsjail/nsjail /bin/nsjail
+
+COPY --from=ghcr.io/decompals/wibo:0.6.16 /usr/local/sbin/wibo /usr/bin/
+
+# Patched mips binutils
+RUN wget "https://github.com/decompals/binutils-mips-ps2-decompals/releases/download/v0.4/binutils-mips-ps2-decompals-linux-x86-64.tar.gz" \
+    && tar xvzf binutils-mips-ps2-decompals-linux-x86-64.tar.gz -C /usr/bin mips-ps2-decompals-as mips-ps2-decompals-nm mips-ps2-decompals-objdump \
+    && rm binutils-mips-ps2-decompals-linux-x86-64.tar.gz \
+    && chmod +x /usr/bin/mips-ps2-decompals-*
+
+# Patched PowerPC binutils
+RUN curl -sSL "https://github.com/encounter/gc-wii-binutils/releases/download/2.42-1/linux-x86_64.zip" | \
+    bsdtar -xvf- -C /usr/bin \
+    && chmod +x /usr/bin/powerpc-eabi-*
+
+# MSDOS specific
+RUN wget "https://github.com/OmniBlade/binutils-gdb/releases/download/omf-build/omftools.tar.gz" \
+    && tar xvzf omftools.tar.gz -C /usr/bin jwasm \
+    && rm omftools.tar.gz \
+    && wget "https://github.com/decompals/binutils-omf/releases/download/v0.4/omftools-linux-x86_64.tar.gz" \
+    && tar xvzf omftools-linux-x86_64.tar.gz -C /usr/bin omf-nm omf-objdump \
+    && rm omftools-linux-x86_64.tar.gz
+
+RUN mkdir -p /etc/fonts
+
+ENV WINEPREFIX=/tmp/wine
+
+# Ensure /sandbox and wine dirs have correct ownership
+RUN mkdir -p /sandbox \
+    && chown -R ubuntu:ubuntu /sandbox \
+    && mkdir -p "${WINEPREFIX}" \
+    && chown ubuntu:ubuntu "${WINEPREFIX}"
+
+# Switch to non-root user
+USER ubuntu
+
+# Initialize wine files to /home/ubuntu/.wine
+RUN wineboot --init
+
+# Create cromper user
+RUN useradd -m cromper
+
+# Create necessary directories
+RUN mkdir -p /cromper/compilers /cromper/libraries /tmp/sandbox/root /tmp/wine \
+    && chown -R cromper:cromper /cromper /tmp/sandbox /tmp/wine
+
+# Set working directory
+WORKDIR /cromper
+
+# Copy project files
+COPY --chown=cromper:cromper pyproject.toml uv.lock* ./
+COPY --chown=cromper:cromper cromper/ ./cromper/
+COPY --chown=cromper:cromper tests/ ./tests/
+COPY --chown=cromper:cromper conftest.py ./
+COPY --chown=cromper:cromper .env* ./
+
+# Switch to cromper user
+USER cromper
+
+# Install dependencies
+RUN uv sync --frozen
```

Comment:
I don't think we'd do this here, think we'd do it as part of the dev entrypoint.sh